Documentation/ldap-connector.md

```diff
@@ -0,0 +1,86 @@
+# Overview of the `ldap` Connector
+
+The `ldap` connector allows email/password based authentication hosted by dex, backed by a LDAP directory. All the configuration parameters required by the `ldap` connector are held in [Config struct](../connector/ldap/ldap.go).
+
+The connector executes two primary queries:
+
+1. Finding the user based on the end user's credentials.
+2. Searching for groups using the user entry.
+
+User entries are expected to have an email attribute (configurable through "emailAttr"), and a display name attribute (configurable through "nameAttr").
+
+### Setting the Configuration
+
+To set the `ldap` connector configuration in dex, create a config file at `example/config-ldap.yaml`, then use the following command to start an instance of dex:
+
+```
+./bin/dex serve examples/config-dev.yaml
+```
+
+### Example: User authentication
+
+The following is an example config file that can be used by the `ldap` connector to submit a [SearchRequest](https://godoc.org/gopkg.in/ldap.v2#SearchRequest) to authenticate a user.
+
+```yaml
+
+  connectors:
+   - type: ldap
+     config:
+       # host and port of the LDAP server in form "host:port".
+       host: ldap.example.com:636
+       # The following field is required if using port 389.
+       # insecureNoSSL: true
+       # Default: use the host's root CA set.
+       rootCA: /etc/dex/ldap.ca
+       # DN to bind as for search operations.
```

Comment:
Note that these are optional and not required if your LDAP server provides sufficient access for anonymous auth.